code/datums/components/payment.dm

```diff
@@ -17,29 +17,36 @@
 	var/transaction_style = "Clinical"
 	///Who's getting paid?
 	var/datum/bank_account/target_acc
+	///Does this payment component respect same-department-discount?
+	var/department_discount = FALSE
 
 /datum/component/payment/Initialize(_cost, _target, _style)
 	target_acc = _target
 	if(!target_acc)
 		target_acc = SSeconomy.get_dep_account(ACCOUNT_CIV)
-
 	cost = _cost
 	transaction_style = _style
 	RegisterSignal(parent, COMSIG_OBJ_ATTEMPT_CHARGE, .proc/attempt_charge)
 	RegisterSignal(parent, COMSIG_OBJ_ATTEMPT_CHARGE_CHANGE, .proc/change_cost)
 
 /datum/component/payment/proc/attempt_charge(datum/source, atom/movable/target, extra_fees = 0)
 	SIGNAL_HANDLER
-
-	if(!cost) //In case a free variant of anything is made it'll skip charging anyone.
+	if(!cost && !extra_fees) //In case a free variant of anything is made it'll skip charging anyone.
 		return
+	var/total_cost = cost + extra_fees
 	if(!ismob(target))
 		return COMPONENT_OBJ_CANCEL_CHARGE
 	var/mob/living/user = target
+	if(issilicon(user)) //They have evolved beyond the need for mere credits
+		return
 	var/obj/item/card/id/card
 	if(istype(user))
 		card = user.get_idcard(TRUE)
+	if(!card && istype(user.pulling, /obj/item/card/id))
+		card = user.pulling
 	if(!card)
+		if(handle_cardless(user, total_cost)) //Here we attempt to handle the purchase physically, with held money first. Otherwise we default to below.
```

Comment:
should everything that uses this component accept cardless?

Reply:
Currently, anyway, I don't see any reason why not. However things down the line may want that I suppose.